GLOSSARY.md

```diff
@@ -0,0 +1,96 @@
+These are some terms and structures frequently used throughout the zed codebase. This is a best effort list.
+
+Questions:
+
+- Can we generate this list form doc comments throughout zed?
+- We should have a section that shows the various UI parts and their names. (Cant do that in channel)
+
+## General
+- `Any<thing>`: A type erased version of thing. Think Box<dyn Trait>.
+- `<Thing>Store`: A wrapper type which abstracts over whether operations are running locally or on a remote.
+
+## GPUI
+### State menagement
+- `App`: A singleton which holds the full application state including all the entities. Crucially: App is not Send, which means that App only exists on the thread that created it (which is the main/UI thread, usually). Thus, if you see a &mut App, know that you're on UI thread.
+- `Context`: A wrapper around the App struct with specialized behavior for a specific Entity. Think of it as `(&mut App, Entity<V>)`. The specialized behavior is surfaced in the API surface of Context. E.G App::spawn takes a `AsyncFnOnce(AsyncApp) -> Ret`, whereas Context::spawn takes a `AsyncFnOnce(WeakEntity<V>, AsyncApp) -> Ret`.
+- `AsyncApp`: An owned version of App for use in async contexts. This type is *still* not Send (so AsyncApp = you're on the main thread) and any use of it may be fallible (to account for the fact that the App might've been terminated by the time this closure runs).
```

Comment:
```suggestion
- `AsyncApp`: An owned version of `App` for use in async contexts. This type is *still* not `Send` (so `AsyncApp` = you're on the main thread) and any use of it may be fallible (to account for the fact that the `App` might've been terminated by the time this closure runs).
```